lib/amazon/redshift.rb

```diff
@@ -169,9 +169,11 @@ def _sql_drop_table_statement
         "DROP TABLE #{@schema}.#{@table_name} CASCADE"
       end
 
-      def _sql_create_table_as_statement(sql_string)
-        "CREATE TABLE #{@schema}.#{@table_name} AS
-        #{sql_string}"
+      def _sql_create_table_as_statement(sql_string, table_attributes = nil)
+        sql = ["CREATE TABLE #{@schema}.#{@table_name}"]
+        sql << table_attributes unless table_attributes.nil?
+        sql << "AS #{sql_string}"
+        sql.join(" ")
```

Comment:
Since you are using an array anyway, consider using `compact`:

```rb
def _sql_create_table_as_statement(sql_string, table_attributes = nil)
  ["CREATE TABLE #{@schema}.#{@table_name}", table_attributes, "AS #{sql_string}"].compact.join(" ")
end
```